src/routes/TableView.tsx

```diff
@@ -1,6 +1,104 @@
 import * as React from 'react';
 import Typography from '@material-ui/core/Typography';
+import IconButton from '@material-ui/core/IconButton';
+import SettingsIcon from '@material-ui/icons/Settings';
+import Table from '@material-ui/core/Table';
+import { Paper, TableBody, TableCell, TableHead, TableRow } from '@material-ui/core';
+import { TableSettingsModal } from '../components/TableSettingsModal';
+import { flatten, has, keys, map, mapKeys } from 'lodash';
+import { Teams } from '../data/team';
+import { calculations } from '../data/calculations';
+import { connect } from 'react-redux';
+import { AppState } from '../state/state';
+import { take } from '../utils';
 
-export const TableView = () => (
-    <Typography variant="display2">Table View</Typography>
-);
+export const TableView = connect(
+    (state: AppState) => ({ teams: state.teams })
+)(
+    class extends React.Component<TableViewProps, TableViewState> {
+    state: TableViewState = {
+        isModalOpen: false,
+        gameName: '',
+        calculationSettings: []
+    };
+
+    configureTable = (gameName: string, calculationSettings: Array<CalculationSetting>) => {
+        this.setState({ gameName, calculationSettings });
+    };
+
+    handleModalOpen = () => {
+        this.setState({ isModalOpen: true });
+    };
+
+    handleModalClose = () => {
+        this.setState({ isModalOpen: false });
+    };
+
+    render() {
+        const { teams } = this.props;
+        const { isModalOpen, calculationSettings } = this.state;
+        return (
+            <div>
+                <Typography variant="display2">Table View</Typography>
+                <TableSettingsModal
+                    configureTable={this.configureTable}
+                    isModalOpen={isModalOpen}
+                    handleModalClose={this.handleModalClose}
+                />
+                <IconButton onClick={this.handleModalOpen}>
+                    <SettingsIcon/>
+                </IconButton>
+                <Paper>
```

Comment:
```suggestion
                <Paper style={{overflowX: 'scroll'}}>
```

or however you like to do your styles.